docs-content: All docs changes for v2

oponder opened the pull request:
Gathering place to review and comment on changes being made on docs-content due to v2 service definitions.

kordless commented on content/guides/annotated-helloworld.md:
```diff
@@ -50,7 +50,7 @@ You will be prompted to enter your Giant Swarm password. Note that the password
 ## Getting the code
 
-We provide a very simple [example application](https://github.com/giantswarm/helloworld) for you on GitHub. To checkout the source via git, do
+We provide the source for our Hellow World example on [GitHub](https://github.com/giantswarm/helloworld). To checkout the source via git, do
```
Hello World, not Hellow World

kordless commented on content/guides/annotated-helloworld.md:
````diff
@@ -61,74 +61,68 @@ Alternatively you can [download a ZIP file](https://github.com/giantswarm/hellow
 ```
 
-Now your application is created on Giant Swarm. The `python:3` image you are using in your component is loaded from the Docker registry and started as a container on some machine of our cluster. Access to the application via your configured domain name is set up.
+Now your service is created on Giant Swarm. The image you are using in your component is loaded from the Docker registry and started as a container on some machine on our cluster. Access to the service via your configured domain name is set up a moment after the service is launched.
````
"started as a container on *a* machine" and "Access to the service is done via the domain used in the configuration."

Figure as long as we are cleaning things up...

kordless commented on content/reference/cli/exec.md:
```diff
@@ -1,14 +1,14 @@
 +++
 
-# `swarm exec`: Run Command on Instance
+# `swarm exec`: Run Command on an Instance
```
In English, you can leave an article (a/an) out of the sentence if you are talking about a generalized case. In this case, the 'instance' is generalized, so you can leave it out.

Or you can put it in and make the header line break on the page, which is why I took it out. :)

kordless commented on content/reference/cli/logs.md:
```diff
@@ -68,5 +68,5 @@ swarm logs <instance-id> -f
- * [Getting a List of Instances for an Application](/reference/cli/status/)
- * [Getting an Application's Statistics](/reference/cli/stats/)
+ * [Getting a List of Instances for a Service](/reference/cli/status/)
+ * [Getting a Services' Statistics](/reference/cli/stats/)
```
"a Service's" because 'service' can be used in a singular fashion without an S. You only append ' to words that have an 'S' on both the singular and plural forms.  Example "Bob tore off several car's bumpers when he was parking his tank in the lot."

kordless commented on content/reference/cli/ls.md:
```diff
@@ -30,23 +30,23 @@ __Example with Response__
- * `environment` - The name of the environment in which the application is running.
- * `created` - The application creation datetime in UTC.
- * `status` - The status of the application.
+ * `service` - The service name as defined upon creation.
```
How about "as defined in the definition"?

oponder replied:
It's possible now leave out the name in the definition and give a name while creating the app.

kordless commented on content/reference/cli/ls.md:
```diff
@@ -30,23 +30,23 @@ __Example with Response__
 ## Status Explained
 
-The application status is an aggregate status of of all component instances within an application. The possible statuses are:
+The service status is an aggregate status of of all component instances within a service. The possible statuses are:
```
Change "The possible statuses are" to "Status may be one of the following:".  The use of plurals with status is confusing, at best.

kordless commented on content/reference/cli/scaledown.md:
```diff
@@ -10,14 +10,14 @@ weight = 91
 ## Command Syntax
 
-The `swarm scaledown` command is called using service component name and the number of instances to remove form the component as arguments. If a number of instances is not given, the number of instances is scaled down by one (1) instance.
+The `swarm scaledown` command is called using a component name and the number of instances to remove from the component as arguments. If a number of instances is not given, the number of instances is scaled down by one (1) instance.
```
That was me! :)

oponder replied:
*highfive* :D

kordless commented on content/reference/cli/start.md:
```diff
@@ -1,96 +1,102 @@
+# `swarm start`: Start Service or Component
 
-The `swarm start` command is used to start an application or a service inside an application after it has been created using the [`swarm create`](/reference/cli/create/) command.
+The `swarm start` command is used to start a service or a component inside a service after it has been created using the [`swarm create`](/reference/cli/create/) command.
```
"used to start a service, or a component inside a service, after it" Commas.

kordless commented on content/reference/cli/start.md:
```diff
@@ -1,96 +1,102 @@
+The `swarm start` command is used to start a service or a component inside a service after it has been created using the [`swarm create`](/reference/cli/create/) command.
 
-__Note:__ To start the applicaiton, the `swarm start` command will need to be using the same organization and environment used by the `swarm create` command to create the application.
+__Note:__ To start the service, the `swarm start` command will need to be using the same organization and environment used by the `swarm create` command when you created the service.
```
"to create the service". I'm trying to remove 'you' and 'your' from the docs.

kordless commented on content/reference/cli/status.md:
```diff
@@ -1,27 +1,40 @@
 ## Command Syntax
 
-The `swarm status` command is called using the application name, which is defined in the `swarm.json` file using the `app_name` key when you start the application.
+The `swarm status` command is called using the service name, which is defined in the `swarm.json` file using the `name` key when you start the service.
```
"when a service is started". Trying to remove 'you' and 'your' from the docs.

kordless commented on content/reference/cli/status.md:
````diff
@@ -1,27 +1,40 @@
+swarm status <service_name>/<component>
+```
+
+In this case `swarm status` only returns the status of instances of that component. If you'd also like to see the status of child components use the `-r` or `--recursive` flag:
````
"To see the status of child.." Again, removing 'you' from docs.

kordless commented on content/reference/cli/env.md:
````diff
@@ -71,7 +71,7 @@ $ swarm env -d startup/test
 ```
 
-__Note:__ Deleting an environment does not delete or stop any applications deployed in the environment's scope. This means it is entirely possible to 'strand' applications, where you can't stop or delete them. Be careful out there!
+__Note:__ Deleting an environment does not delete or stop any services deployed in the environment's scope. This means it is entirely possible to 'strand' services, where you can't stop or delete them. Be careful out there!
````
I missed this 'you'.  Change to "where they can't be stopped or deleted."

kordless commented on content/reference/cli/stop.md:
```diff
@@ -1,58 +1,64 @@
+When a service is stopped with `swarm stop`, all component instances are stopped. 
 
-The command can be called to stop an entire application without any arguments if a valid `swarm.json` file is present in the current working directory and represents a running application with the same name, in the current organization and environment as reported by `swarm env`:
+The command can be called to stop an entire service without any arguments if a valid `swarm.json` file is present in the current working directory that represents a running service with the same name, in the current organization and environment as reported by `swarm env`:
```
"working directory *which* represents"

kordless commented on content/reference/cli/up.md:
```diff
@@ -1,37 +1,37 @@
 ## Command Syntax
 
-The `swarm up` command can be called to create and start an application without any arguments if a valid `swarm.json` file is present in the current working directory:
+The `swarm up` command can be called to create and start a service without any arguments if a valid `swarm.json` service definition file containing a service name is present in the current working directory:
```
Is a service definition file able to be valid without a service name? If so, this is fine.

oponder replied:
yes, we support assigning a name randomly or via the `-n` flag of `swarm create`

kordless commented on content/reference/cli/update.md:
```diff
@@ -22,17 +22,17 @@ __Note:__ A component's instance will be started if it was not running before th
 ## Command Syntax
 
-The `swarm update` command is called using the application, service and component name to update the component image with the latest image:
+The `swarm update` command is called using the service and component name to update the component image with the latest image:
```
"service and component *names* to"

kordless commented:
I'm all done reviewing!

kordless commented:
Actually, found this also needs to be updated @oponder: https://github.com/giantswarm/docs-content/blob/servicedef-v2/content/reference/swarm-json.md

oponder commented:
@kordless: 
Thanks for the reviews, I'll apply the changes now.

As for the swarm-json reference, @marians is on it here: https://github.com/giantswarm/docs-content/pull/121

By the way, I don't feel insulted if you fix small typos for me :) But I'm glad to fix them also based on comments if you've got a lot of momentum doing reviews and don't want to break the flow.

oponder commented:
@kordless: Done applying the feedback. I also found a couple of "you"s  here and there which I adjusted. Haven't comprehensively gone through it all, just fixed it where I found it for now.

marians commented on content/reference/api/index.md:
```diff
@@ -771,95 +775,98 @@ curl -sS \
-### Starting a Service {#StartService}
+In addition to the path parameters shown above, the following parameters are used inside the JSON `POST` on various methods:
+
+| Key | Value Type | Description | Required by related methods? |
```
@kordless This combined table and "Required by related methods" seems pretty irritating. I'd prefer to describe the JSON data to be posted per method. This way, it's hard to tell apart what is needed for which method.
